orangecontrib/text/twitter.py

```
@@ -0,0 +1,121 @@
+        self.api = credentials.construct_api()
+        self.stream = tweepy.Stream(auth=self.api.auth, listener=Listener(self))
+        self.statuses_lock = Lock()
+
+        self.container = []
+
+    # TODO: add arguments
+    def search(self, *, word_list, max_tweets, lang=None):
+        if not word_list:
+            raise ValueError("Word list can't be empty.")
+
+        query = ' OR '.join(word_list)
+
+        for status in tweepy.Cursor(self.api.search, q=query, lang=lang).items(max_tweets):
+            self.add_status(status)
+
+    def listen(self, *, word_list):
+        self.stream.filter(track=word_list, async=True)
+
+    def add_status(self, status):
+        status_record = {field: getattr(status, field)
+                         for field in self.supported_fields}
+        self.statuses_lock.acquire()
+        self.container.append(status_record)
+        self.statuses_lock.release()
+        self.on_status()
+
+    def create_corpus(self):
+        """Creates corpus from collected tweets.
+        """
+        X = np.array([
+            [float(record[field]) for field in self.continuous_fields]
+            for record in self.container
+        ])
+
+        metas = np.array([
+             [record[field] for field in self.string_fields]
+             for record in self.container
+        ], dtype=object)
+
+        metas_attr = [data.StringVariable(name) for name in self.string_fields]
+        domain = data.Domain(attributes=[data.ContinuousVariable(name) for name in self.continuous_fields],
+                             metas=metas_attr)
+
+        return Corpus(X=X, metas=metas, domain=domain, text_features=metas_attr)
+
+    def on_status(self):
+        pass
+
+
+class Listener(tweepy.StreamListener):
+    def __init__(self, api):
+        super().__init__()
+        self.api = api
+
+    def on_status(self, status):
+        self.api.add_status(status)
+
+    def on_error(self, status):
+        print(status)
```

This print should be removed in the final version. If there is an error it should be propagated to widget instead of printing into console.